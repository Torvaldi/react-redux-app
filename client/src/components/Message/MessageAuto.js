import React, { Fragment } from 'react';
import propTypes from 'prop-types';

const MessageAuto = ({username, message, messageClass}) => {

    return(
        <Fragment>
            <section className="messageUsernameLayout">
                <span className={messageClass}>{username} {message}</span>
            </section>
        </Fragment>
    );

};

MessageAuto.propTypes = {
    username: propTypes.string.isRequired,
    message: propTypes.string.isRequired,
}

export default MessageAuto
