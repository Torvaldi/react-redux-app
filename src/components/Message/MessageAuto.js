import React, { Fragment } from 'react';
import propTypes from 'prop-types';

const MessageAuto = ({username, message, messageClass}) => {

    return(
        <Fragment>
            <section className="messageUsernameLayout">
                <span className={messageClass}>{username} {message}</span>
            </section>
        </Fragment>
    );

};

MessageAuto.propTypes = {
    username: propTypes.string.isRequired,
    findAnime: propTypes.bool.isRequired,
}

export default MessageAuto
