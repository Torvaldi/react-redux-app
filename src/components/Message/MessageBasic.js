import React, { Fragment } from 'react';
import propTypes from 'prop-types';

const MessageBasic = ({usernameClass, username, message}) => {

    return(
        <Fragment>
            <section className="messageUsernameLayout">
                <span className={usernameClass}>{username}</span>
            </section>
            <section>
                <p className="messageContent">{message}</p>
            </section>
        </Fragment>
    );

};


MessageBasic.propTypes = {
    authUser: propTypes.object.isRequired,
    user: propTypes.object.isRequired,
    message: propTypes.string.isRequired
}


export default MessageBasic
