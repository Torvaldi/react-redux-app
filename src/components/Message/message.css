
.messageLayout {
    list-style-type: none;
    /*
    background-color: rgba(0, 0, 0, 0.3);
    border-bottom: 1px solid #212121;
    */
    padding: 10px 15px;
}

.messageUsernameLayout {
}

.messageContentLayout {
}

.messageUsername, .messageAuthUsername {
    font-weight: bold;
    color: white;
   
}

.messageUsername {
    /**/
}

.messageAuthUsername {
    color: #212121;
}

.messageContent{
    margin: 5px 0 0 0;
    color: white;
    padding-bottom: 10px;
    border-bottom: 1px solid rgba(255, 255, 255, 0.1);
}

.messageAutoRight {
    color: #8bc34a
}

.messageAutoWrong {
    color: red;
}

.messageAutoWrong, .messageAutoRight {
    font-style: italic;
}
