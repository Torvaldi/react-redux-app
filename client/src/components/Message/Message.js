import React from 'react';
import propTypes from 'prop-types';
import './message.css';

import MessageBasic from './MessageBasic';
import MessageAuto from './MessageAuto';

import { messageType as messageTypeHelper } from '../../helper/chat';

const Message = ({username, message, messageType, authUser}) => {

  // ! user write a message
  if(messageType === messageTypeHelper.MESSAGE){
    let usernameClass = 'messageUsername';

    if(authUser.username === username){
      usernameClass = 'messageAuthUsername'
    }

    return(
      <li className="messageLayout">
        <MessageBasic usernameClass={usernameClass} username={username} message={message} />
      </li>
    );

  // ! user leave the game
  } else if(messageType === messageTypeHelper.USER_LEAVE){

    return(
      <li className="messageLayout">
        <MessageAuto username={username} message=" left the game" messageClass="messageAutoContent" />
      </li>
    );

  } else if(messageType === messageTypeHelper.USER_JOIN){
 
    return(
      <li className="messageLayout">
        <MessageAuto username={username} message=" join the game" messageClass="messageAutoContent" />
      </li>
    );

  // ! user answer 
  } else {

    return(
      <li className="messageLayout">
        <MessageAuto username={username} message=" chose an answer" messageClass="messageAutoContent" />
      </li>
    );

  }


}


Message.propTypes = {
  authUser: propTypes.object.isRequired,
 // username: propTypes.string.isRequired,
  message: propTypes.string,
  messageType: propTypes.string.isRequired
}


export default Message
