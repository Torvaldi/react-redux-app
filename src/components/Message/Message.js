import React from 'react';
import propTypes from 'prop-types';
import './message.css';

import MessageBasic from './MessageBasic';
import MessageAuto from './MessageAuto';

import { messageType as messageTypeHelper } from '../../helper/chat';

const Message = ({username, message, messageType, authUser}) => {
  
  // ! user write a message
  if(messageType === messageTypeHelper.MESSAGE){
    let usernameClass = 'messageUsername';

    if(authUser.username === username){
      usernameClass = 'messageAuthUsername'
    }

    return(
      <li className="messageLayout">
        <MessageBasic usernameClass={usernameClass} username={username} message={message} />
      </li>
    );

  // ! user leave the game
  } else if(messageType === messageTypeHelper.USER_LEAVE){

    return(
      <li className="messageLayout">
        <MessageAuto username={username} message=" left the game" messageClass="" />
      </li>
    );

  } else if(messageType === messageTypeHelper.USER_JOIN){
 
    return(
      <li className="messageLayout">
        <MessageAuto username={username} message=" join the game" messageClass="" />
      </li>
    );

  // ! user answer 
  } else {

    return(
      <li className="messageLayout">
        <MessageAuto username={username} message=" chose an answer"  />
      </li>
    );

  }


}


Message.propTypes = {
  authUser: propTypes.object.isRequired,
  username: propTypes.string.isRequired,
  message: propTypes.string,
  messageType: propTypes.string.isRequired
}


export default Message
