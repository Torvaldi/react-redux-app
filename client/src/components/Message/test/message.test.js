import React from 'react';
import { shallow, configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Message from '../Message';

configure({ adapter: new Adapter() });

describe('<Message />', () => {
    // snapshot
    it('should render Message Componenent', () => {
        let authUser = {id:1}
        let user = {id: 1, username : "bob"}
        let message = 'Bonjour';

        const wrapper = shallow(<Message authUser={authUser} user={user} message={message} />);
        expect(wrapper).toMatchSnapshot();
    });
    
    it('should render with props', () => {
        let authUser = { id: 1}
        let user = {id: 1, username : "bob"}
        let message = 'Bonjour';
        const messageCompenent = mount(<Message authUser={authUser} user={user} message={message} />);
        
        expect(messageCompenent.prop('authUser')).toEqual(authUser);
        expect(messageCompenent.prop('user')).toEqual(user); 
        expect(messageCompenent.prop('message')).toEqual(message); 
    });
    
});