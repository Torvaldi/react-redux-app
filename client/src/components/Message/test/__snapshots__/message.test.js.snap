// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> should render Message Componenent 1`] = `ShallowWrapper {}`;
